dali/operators/random/beta_distribution.h

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_RANDOM_BETA_DISTRIBUTION_H_
+#define DALI_OPERATORS_RANDOM_BETA_DISTRIBUTION_H_
+
+#include <random>
+#include <vector>
+#include "dali/operators/random/rng_base.h"
+#include "dali/pipeline/operator/arg_helper.h"
+
+#define DALI_BETA_DIST_TYPES float, double
+
+namespace dali {
+
+template <typename T>
+struct BetaDistributionImpl {
+  BetaDistributionImpl() : BetaDistributionImpl{1, 1} {}
+
+  explicit BetaDistributionImpl(T alpha, T beta)
+      : has_small_param_{alpha < 1 && beta < 1},
+        alpha_{alpha},
+        beta_{beta},
+        b_div_a_{beta_ / alpha_},
+        a_{has_small_param_ ? alpha_ + 1 : alpha_},
+        b_{has_small_param_ ? beta_ + 1 : beta_},
+        exp_{} {}
+
+  template <typename Generator>
+  T Generate(Generator &st) {
+    return has_small_param_ ? GenerateGammaExp(st) : GenerateGamma(st);
+  }
+
+ private:
+  template <typename Generator>
+  T GenerateGamma(Generator &st) {
+    // https://en.wikipedia.org/wiki/Beta_distribution#Random_variate_generation
+    T a = a_(st);
+    T b = b_(st);
+    return a / (a + b);
+  }
+
+  template <typename Generator>
+  T GenerateGammaExp(Generator &st) {
```

Comment:
A general comment: this is very compute-intensive - you're drawing multiple values from non-trivial distributions. I wonder what would be the threshold number of numbers to obtain at which point it'd be cheaper to run inverse sampling (which would require some precalculation).